equinox/nn/pool.py

```diff
@@ -0,0 +1,238 @@
+from typing import Sequence, Tuple, Union, Callable, Optional
+import einops
+
+import jax.lax as lax
+import jax.numpy as jnp
+import jax.random
+import numpy as np
+
+from ..custom_types import Array
+from ..module import Module, static_field
+
+
+class Pool(Module):
+    """ General N-dimensional downsampling over a sliding window. """
+
+    operation: Callable[[Array], Array] = static_field()
```

Comment:
This shouldn't be static. (In principle the operation may be learnt.) Also I think the annotation should be `Callable[[Array, Array], Array]`.